File: C_program4/py/ber.py
import const

total_error = 0

def ber(loop, transmit_bit, receive_bit, f, CNR):
    for i in range(const.BITN):
        if transmit_bit[i] != receive_bit[i]:
            total_error += 1
    
    if loop == (const.LOOPN - 1):
        Pe = total_error / (const.LOOPN * const.BITN)
        print("Eb/N0 = %f, Average BER = %f" % 
            ((CNR - 3.0), Pe))
        f.write("Eb/N0 = %f, Average BER = %f" % 
            ((CNR - 3.0), Pe))
        total_error = 0

File: C_program4/CppVer/src/ber.cpp
#include "const.h"

void ber(int loop, int (*tbit), int (*rbit), FILE *fp, double CNR)
{
	int n, error = 0;
	static double error_count = 0.0;

	for (n = 0; n <  BITN; n++)
	{
		if (tbit[n] != rbit[n])
			error++;
	}

	error_count += (double)error;

	if(loop == LOOPN-1)
	{
		error_count /= (LOOPN * BITN);
		printf("Eb/N0 = %f, Average BER = %1.10f\n", (CNR - 3.0), error_count);
		fprintf(fp, "%f\t%1.10f\n", (CNR - 3.0), error_count);
		error_count = 0.0;
	}
}
